File: geo-project/geo-web/src/api/presaleBenchmark.ts
import request from './request'
import type { R } from '@/types'

export interface PresaleBenchmark {
  id: number
  industry: string
  industryRole: string
  avgOverall: number
  avgMention: number
  avgRanking: number
  avgSentiment: number
  avgCoverage: number
  top1Overall: number
  top1Mention: number
  top1Ranking: number
  top1Sentiment: number
  top1Coverage: number
  top10Score: number
  confidenceLevel: string
  source: string
  sampleSize: number
  enabled: boolean
  effectiveFrom: string
  remark?: string
  createdAt?: string
  updatedAt?: string
}

export type PresaleBenchmarkPayload = Omit<PresaleBenchmark, 'id' | 'source' | 'createdAt' | 'updatedAt'>

export interface PresaleBenchmarkHistory {
  id: number
  benchmarkId: number
  industry: string
  industryRole: string
  operation: string
  beforeSnapshot?: string
  afterSnapshot?: string
  operatorName?: string
  remark?: string
  createdAt: string
}

export function listPresaleBenchmarks() {
  return request.get<R<PresaleBenchmark[]>>('/presale/benchmarks')
}

export function createPresaleBenchmark(data: PresaleBenchmarkPayload) {
  return request.post<R<PresaleBenchmark>>('/presale/benchmarks', data)
}

export function updatePresaleBenchmark(id: number, data: PresaleBenchmarkPayload) {
  return request.put<R<PresaleBenchmark>>(`/presale/benchmarks/${id}`, data)
}

export function updatePresaleBenchmarkStatus(id: number, enabled: boolean, remark?: string) {
  return request.put<R<PresaleBenchmark>>(`/presale/benchmarks/${id}/status`, { enabled, remark })
}

export function listPresaleBenchmarkHistory(benchmarkId?: number) {
  return request.get<R<PresaleBenchmarkHistory[]>>('/presale/benchmarks/history', {
    params: benchmarkId ? { benchmarkId } : undefined,
  })
}

File: geo-project/geo-server/src/main/java/com/huanjing/geo/module/presale/persist/entity/PresaleBenchmark.java
package com.huanjing.geo.module.presale.persist.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("presale_benchmark")
public class PresaleBenchmark {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String industry;
    private String industryRole;
    private BigDecimal avgOverall;
    private BigDecimal avgMention;
    private BigDecimal avgRanking;
    private BigDecimal avgSentiment;
    private BigDecimal avgCoverage;
    private BigDecimal top1Overall;
    private BigDecimal top1Mention;
    private BigDecimal top1Ranking;
    private BigDecimal top1Sentiment;
    private BigDecimal top1Coverage;
    private BigDecimal top10Score;
    private String confidenceLevel;
    private String source;
    private Integer sampleSize;
    private Boolean enabled;
    private LocalDate effectiveFrom;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
